File: bin/check_db.py
import sqlite3
import os

db_path = os.path.expandvars(r'%USERPROFILE%\Documents\supanotes.sqlite')
print(f"Checking DB at {db_path}")

if not os.path.exists(db_path):
    print("DB file not found!")
    exit(1)

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

cursor.execute("SELECT COUNT(*) FROM notes")
print(f"Total notes: {cursor.fetchone()[0]}")

cursor.execute("SELECT id, content, deleted_at FROM notes WHERE trim(content) = '' AND deleted_at IS NULL")
empty_notes = cursor.fetchall()
print(f"Notes with empty content (not deleted): {len(empty_notes)}")

for note_id, content, deleted_at in empty_notes:
    cursor.execute("SELECT COUNT(*) FROM note_nodes WHERE note_id = ?", (note_id,))
    node_count = cursor.fetchone()[0]
    print(f"  - Note ID: {note_id}, nodes: {node_count}")

conn.close()

File: bin/check_db.dart
import 'dart:io';
import 'package:sqlite3/sqlite3.dart';

void main() {
  final dbPath = '${Platform.environment['USERPROFILE']}\\Documents\\supanotes.sqlite';
  print('Checking DB at $dbPath');
  
  if (!File(dbPath).existsSync()) {
    print('DB not found!');
    return;
  }
  
  final db = sqlite3.open(dbPath);
  
  // Count total notes
  final totalNotes = db.select('SELECT COUNT(*) as c FROM notes').first['c'];
  print('Total notes: $totalNotes');
  
  // Notes missing from the list (content is empty)
  final missingNotes = db.select("SELECT id, content FROM notes WHERE trim(content) = ''");
  print('Notes with empty content: ${missingNotes.length}');
  
  for (final row in missingNotes) {
    print('  - Note ID: ${row['id']}, Content length: ${row['content'].toString().length}');
    
    // Check if they have node_nodes
    final nodes = db.select("SELECT count(*) as c FROM note_nodes WHERE note_id = ?", [row['id']]).first['c'];
    print('    - Has $nodes note_nodes');
  }
  
  db.dispose();
}
